File: langs/python3/problems/inventory/stub.py
class Simulation:
    def __init__(self):
        pass

    def create_item(self, sku, name):
        """Returns "true" if created, "false" if the sku exists."""
        raise NotImplementedError

    def stock(self, sku, delta):
        """Add delta units. New qty as a string, "" if missing, or invalid_request."""
        raise NotImplementedError

    def get_qty(self, sku):
        """On-hand quantity as a string, or ""."""
        raise NotImplementedError

    def list_low(self, threshold):
        """Skus at or below threshold as id(qty), or ""."""
        raise NotImplementedError

    def reserve(self, sku, n):
        """Hold n on-hand units. "true" or "invalid_request"."""
        raise NotImplementedError

    def release(self, sku, n):
        """Free n reserved units. "true" or "invalid_request"."""
        raise NotImplementedError

    def ship(self, sku, n):
        """Consume n reserved units. "true" or "invalid_request"."""
        raise NotImplementedError

File: langs/java/problems/inventory/stub.java
public class Simulation {
    public Simulation() {}

    /**
     * Returns {@code "true"} if created, {@code "false"} if the sku exists.
     */
    public String createItem(String sku, String name) {
        return "";
    }

    /**
     * Add delta units. New qty as a string, {@code ""} if missing, or
     * {@code invalid_request}.
     */
    public String stock(String sku, int delta) {
        return "";
    }

    /**
     * On-hand quantity as a string, or {@code ""}.
     */
    public String getQty(String sku) {
        return "";
    }

    /**
     * Skus at or below threshold as {@code id(qty)}, or {@code ""}.
     */
    public String listLow(int threshold) {
        return "";
    }

    /**
     * Hold n on-hand units. {@code "true"} or {@code "invalid_request"}.
     */
    public String reserve(String sku, int n) {
        return "";
    }

    /**
     * Free n reserved units. {@code "true"} or {@code "invalid_request"}.
     */
    public String release(String sku, int n) {
        return "";
    }

    /**
     * Consume n reserved units. {@code "true"} or {@code "invalid_request"}.
     */
    public String ship(String sku, int n) {
        return "";
    }
}
